plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -595,9 +595,18 @@ def _is_retryable(error):
 
 _SANITIZE_LABEL_PATTERN = re.compile(r"[^a-z0-9_-]")
 
+_VALIDATE_LABEL_LENGTH_LIMIT = 63
+
 
 def _sanitize_label(value: str) -> str:
     """Return a legal value for a BigQuery label."""
     value = value.strip().lower()
     value = _SANITIZE_LABEL_PATTERN.sub("_", value)
-    return value
+    value_length = len(value)
+    if value_length > _VALIDATE_LABEL_LENGTH_LIMIT:
+        error_msg = (
+            f"Current label length {value_length} is greater than length limit: {_VALIDATE_LABEL_LENGTH_LIMIT} | Current sanitized label: {value}"
+        )
+        raise Exception(error_msg)
```

Comment:
```suggestion
        raise RuntimeException(error_msg)
```
This way, instead of seeing `Unhandled error while executing model...`, you'll just see `ERROR` and the runtime error reported in the error summary.